Tasks/AzureRmWebAppDeployment/AzureRmWebAppDeployment.ps1

```diff
@@ -1,134 +1,142 @@
+		Write-Verbose "Is empty" -verbose
+		$setParametersFilePath = ""
+	} else {
+		Write-Verbose "Is not empty" -verbose
+		$setParametersFilePath = Get-SingleFilePath -file $SetParametersFile
+	}
 
-# Import all the dlls and modules which have cmdlets we need
-Import-Module "Microsoft.TeamFoundation.DistributedTask.Task.Internal"
-Import-Module "Microsoft.TeamFoundation.DistributedTask.Task.Common"
+	# Get destination azureRM webApp connection details
+	$azureRMWebAppConnectionDetails = Get-AzureRMWebAppConnectionDetails -webAppName $WebAppName -deployToSlotFlag $DeployToSlotFlag `
+																		   -resourceGroupName $ResourceGroupName -slotName $SlotName
 
-# Load all dependent files for execution
-Import-Module ./AzureUtility.ps1 -Force
-Import-Module ./Utility.ps1 -Force
-Import-Module ./FindInstalledMSDeploy.ps1
+	# webApp Name to be used in msdeploy command
+	$webAppNameForMSDeployCmd = Get-WebAppNameForMSDeployCmd -webAppName $WebAppName -deployToSlotFlag $DeployToSlotFlag -slotName $SlotName
 
- # Importing required version of azure cmdlets according to azureps installed on machine
- $azureUtility = Get-AzureUtility
+	# Construct arguments for msdeploy command
+	$msDeployCmdArgs = Get-MsDeployCmdArgs -packageFile $packageFilePath -webAppNameForMSDeployCmd $webAppNameForMSDeployCmd -azureRMWebAppConnectionDetails $azureRMWebAppConnectionDetails -removeAdditionalFilesFlag $RemoveAdditionalFilesFlag `
+										   -excludeFilesFromAppDataFlag $ExcludeFilesFromAppDataFlag -takeAppOfflineFlag $TakeAppOfflineFlag -virtualApplication $VirtualApplication -AdditionalArguments $AdditionalArguments `
+										   -setParametersFile $setParametersFilePath
 
- Write-Verbose  "Loading $azureUtility"
- Import-Module ./$azureUtility -Force
+	# Deploy azureRM webApp using msdeploy Command
+	Run-MsDeployCommand -msDeployExePath $msDeployExePath -msDeployCmdArgs $msDeployCmdArgs
 
-$ErrorActionPreference = 'Stop'
+	# Get azure webapp hosted url
+	$azureWebsitePublishURL = Get-AzureRMWebAppPublishUrl -webAppName $WebAppName -deployToSlotFlag $DeployToSlotFlag `
+																		   -resourceGroupName $ResourceGroupName -slotName $SlotName
 
-#### MAIN EXECUTION OF AZURERM WEBAPP DEPLOYMENT TASK BEGINS HERE ####
+	# Publish azure webApp url
+	Write-Host (Get-VstsLocString -Key "WebappsuccessfullypublishedatUrl0" -ArgumentList $azureWebsitePublishURL)
 
-# Get msdeploy.exe path
-$msDeployExePath = Get-MsDeployExePath
+	# Set ouput vairable with azureWebsitePublishUrl
+	if(-not [string]::IsNullOrEmpty($WebAppUri))
+	{
+	
+		if( [string]::IsNullOrEmpty($azureWebsitePublishURL))
+		{
+			Throw (Get-VstsLocString -Key "Unabletoretrievewebapppublishurlforwebapp0" -ArgumentList $webAppName)
+		}
+	
+		Write-Host "##vso[task.setvariable variable=$WebAppUri;]$azureWebsitePublishURL"
```

Comment:
It would be better to use one of the lib functions to do this, which will also handle escaping. [Write-VstsSetVariable](https://github.com/Microsoft/vsts-task-lib/blob/master/powershell/Docs/FullHelp/Write-VstsSetVariable.md) or Set-[VstsTaskVariable](https://github.com/Microsoft/vsts-task-lib/blob/master/powershell/Docs/FullHelp/Set-VstsTaskVariable.md)

Does this variable actually need to be consumed in a subsequent task?